apps/aecore/lib/aecore/multi_node_test_framework/worker.ex

```diff
@@ -0,0 +1,611 @@
+    Port.command(state[node_name1].process_port, cmd)
+
+    cmd = "Peers.try_connect(%{host: 'localhost', port: #{sync_port}, pubkey: pub_key})\n"
+    Port.command(state[node_name1].process_port, cmd)
+    {:reply, :ok, state}
+  end
+
+  def handle_call({:get_state}, _, state) do
+    {:reply, state, state}
+  end
+
+  def handle_call({:delete_all_nodes}, _, state) do
+    # killing all the processes and closing all of the ports of the nodes
+    Enum.each(state, fn {_, val} ->
+      {:os_pid, pid} = Port.info(val.process_port, :os_pid)
+      Port.close(val.process_port)
+      System.cmd("kill", ["#{pid}"])
+
+      # deleting nodes folders
+      val.path
+      |> String.replace("elixir-node", "")
+      |> File.rm_rf()
+    end)
+
+    {:reply, :ok, %{}}
+  end
+
+  def handle_call({:delete_node, node_name}, _, state) do
+    if Map.has_key?(state, node_name) do
+      # kills the process, closes the port and deletes the node folder
+      {:os_pid, pid} = Port.info(state[node_name].process_port, :os_pid)
+      Port.close(state[node_name].process_port)
+
+      state[node_name].path
+      |> String.replace("elixir-node", "")
+      |> File.rm_rf()
+
+      System.cmd("kill", ["#{pid}"])
+      new_state = Map.delete(state, node_name)
+      {:reply, :ok, new_state}
+    else
+      {:reply, :unknown_node, state}
+    end
+  end
+
+  def handle_call({:ports_info}, _, state) do
+    ports_info = for {name, _} <- state, do: Port.info(state[name].process_port)
+    {:reply, ports_info, state}
+  end
+
+  def handle_call({:compare_nodes_by_registered_oracles, node_name1, node_name2}, _, state) do
+    registered_oracles1 = state[node_name1].registered_oracles
+    registered_oracles2 = state[node_name2].registered_oracles
+
+    if registered_oracles1 == registered_oracles2 do
+      {:reply, :synced, state}
+    else
+      {:reply, :not_synced, state}
+    end
+  end
+
+  def handle_call(
+        {:compare_nodes_by_oracle_interaction_objects, node_name1, node_name2},
+        _,
+        state
+      ) do
+    oracle_interaction_objects1 = state[node_name1].oracle_interaction_objects
+    oracle_interaction_objects2 = state[node_name2].oracle_interaction_objects
+
+    if oracle_interaction_objects1 == oracle_interaction_objects2 do
+      {:reply, :synced, state}
+    else
+      {:reply, :not_synced, state}
+    end
+  end
+
+  def handle_call({:compare_nodes_by_top_block, node_name1, node_name2}, _, state) do
+    block1 = state[node_name1].top_block
+    block2 = state[node_name2].top_block
+
+    if block1 == block2 do
+      {:reply, :synced, state}
+    else
+      {:reply, :not_synced, state}
+    end
+  end
+
+  def handle_call({:new_node, node_name, port, sync_port}, _, state) do
+    cond do
+      Map.has_key?(state, node_name) ->
+        {:reply, :already_exists, state}
+
+      busy_port?(port) || busy_port?(sync_port) ->
+        {:reply, :busy_port, state}
+
+      true ->
```

Comment:
This is good enough, but I think it's probably possible to use some Elixir features to just start multiple nodes with different environment variables (without copying all source code and compilation data).